### DISCUSSION TITLE
Testing

### DISCUSSION THREAD

**Discussion by jackrvan** [jellyfin/jellyfin-web#3725] [Ideas]
Looking at the code I dont see any tests for the jellyfin website. Is there a reason for this? Ive been wanting to learn some more testing frameworks and chose to learn selenium and was thinking of writing tests for jellyfin web in particular since I use it so much. Is this something the team would be interested in or is their a reason their are no tests and I would be wasting my time?

**Comment by thornbill** [comment]
Yes, we would definitely be interested in adding test coverage for jellyfin-web!

The main thing that has prevented us from adding tests is that all of the web client is pretty interdependent, so I think unit tests are just not possible for most of the web client. Although as we refactor code, this should be less true, so we really should start looking at areas that have already been refactored where we _can_ add unit tests.

For all the other places where it's just not feasible to add unit tests today, I had started looking at using Cypress for end-to-end testing. You can see my progress [on this card](https://trello.com/c/PllPiPQs/13-testing) in my roadmap.

**Reply by jackrvan** [reply]
Okay cool. Honestly Even places where we CAN add unit tests its still probably useful to have end to end tests.... more tests is almost never a bad thing haha. Ive never heard of Cypress. I just went with Selenium because its more well known but if youve already done some stuff ill take a look and see if I can build off it
